lib/span.js

```diff
@@ -229,8 +229,12 @@ Span.prototype.run = function (fn) {
  * })
  */
 Span.prototype.runAsync = function (fn) {
-  this.async = true
-  const span = this
+  const span = this;
+  // don't mark the top span async in lambda because it interferes with the display
+  // on the host.
+  if (!(span.topSpan && ao.execEnv.id === 'lambda')) {
+    this.async = true;
```

Comment:
they're the same thing. for consistency i should use span and have changed it now. the reason for `const span = this;` exists is so the span can be referenced in the callback later in the code. i don't recall the history of the edits but most likely `this.async` had been around since the beginning.

`this` is the object the function was called as a method of an object. if there is no explicit object then `this` defaults to the global object (not what you usually want).

```
> x = {bruce: 'says'}
{ bruce: 'says' }
> function xyz () {
... console.log(this);
... }

> xyz.call(x)  // call with x as this.
{ bruce: 'says' }

> x.fn = xyz
[Function: xyz]
> x.fn()
{ bruce: 'says', fn: [Function: xyz] }

> xyz()
Object [global] {
  global: [Circular *1],
  clearInterval: [Function: clearInterval],
  clearTimeout: [Function: clearTimeout],
  setInterval: [Function: setInterval],
  setTimeout: [Function: setTimeout] {
    [Symbol(util.promisify.custom)]: [Function (anonymous)]
  },
  queueMicrotask: [Function: queueMicrotask],
  clearImmediate: [Function: clearImmediate],
  setImmediate: [Function: setImmediate] {
    [Symbol(util.promisify.custom)]: [Function (anonymous)]
  },
  x: { bruce: 'says' },
  xyz: [Function: xyz]
}
```